docs/probe.md

```diff
@@ -0,0 +1,177 @@
+4. Connect `Meter` to `Hub` via `get_hub().connect_meter(meter)`
+
+The examples will show how each of these steps are accomplished.
+
+## Example 1: Model Layer Output
+### User Story
+I am interested in layer output from the second `relu` activation of a `forward` method located in `armory/baseline_models/pytorch/cifar.py`.
+### `Probe` Example Code
+The code below is an example of how to accomplish steps 1 and 2 (note the lines of code with `# added` comments at the end) for a model code that the user is modifying.
+```python
+"""
+CNN model for 32x32x3 image classification
+"""
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from art.estimators.classification import PyTorchClassifier
+
+from armory.instrument import get_probe # added
+probe = get_probe("my_model") # added
+
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+class Net(nn.Module):
+    """
+    This is a simple CNN for CIFAR-10 and does not achieve SotA performance
+    """
+
+    def __init__(self) -> None:
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(3, 4, 5, 1)
+        self.conv2 = nn.Conv2d(4, 10, 5, 1)
+        self.fc1 = nn.Linear(250, 100)
+        self.fc2 = nn.Linear(100, 10)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x.permute(0, 3, 1, 2)  # from NHWC to NCHW
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.conv2(x)
+        x = F.relu(x)
+        x_out = x.detach().cpu().numpy() # added
+        probe.update(layer_output=x_out) # added
+        x = F.max_pool2d(x, 2)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+
+...
+```
+
+
+
+#### Step 1
+After importing `get_probe`, `probe = get_probe("my_model")` creates a `Probe` object with the name `"my_model"`, which is what the user can refer to later to apply processing functions through a `Meter` object.
+
+#### Step 2
+`x_out = x.detach().cpu().numpy()` is taking the layer output of interest (second `relu` activation output) and converting the tensor to `numpy` array on the CPU, which will be passed to `probe`. An updated value of `x_out` is stored in `layer_output` via `probe.update(layer_output=x_out)`. Like the `Probe` name `"my_model"`, `layer_output` can be referenced by the user later to apply additional processing functions through a `Meter` object.
+
+### `Meter` Example Code
+Now that a `Probe` instance has been created, we need to create a `Meter` object to accept any updated values from `Probe` and apply further processing that the user desires. Suppose the user created a script located at `armory/user_init.py`:
+```python
+from armory.instrument import get_hub, Meter
+
+def user_init_function():
+    meter = Meter(
+        "my_layer_identity", lambda x: x, "my_model.layer_output"
```

Comment:
would `"my_arbitrary_meter_name"` be clearer than `"my_layer_identity"`? Same for example 2

Reply:
resolved with 89d200a6